src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxeReferenceImpl.java

```diff
@@ -121,6 +119,12 @@ public PsiElement resolve() {
     return resolve(true);
   }
 
+  public boolean resolveIsExtension() {
+    // @TODO: DIRTY HACK! to avoid rewriting all the code!
```

Comment:
As I noted previously, we don't need the hack if we move/duplicate the test that determines an extension to the models.  Then we can get rid of this hack and the instance variable.